sycl/include/sycl/types.hpp

```diff
@@ -1488,6 +1488,15 @@ class SwizzleOp {
       SwizzleOp<const VecT, GetOp<DataT>, GetOp<DataT>, GetOp, Indices...>;
 
 public:
+  using element_type = DataT;
+
+  const DataT &operator[](int i) const { return (*m_Vector)[i]; }
+
+  template <typename T = VecT>
+  std::enable_if_t<!std::is_const_v<T>, DataT> &operator[](int i) {
```

Comment:
```suggestion
  template <typename _T = VecT>
  std::enable_if_t<!std::is_const_v<_T>, DataT> &operator[](int i) {
```

just to be safe. Also, the spec seems to have them unconditionally. Do we need to update it?

That said, what do we gain by conditionally disabling this?